project/Build.scala

```diff
@@ -2,34 +2,90 @@ import sbt._
 import sbt.Keys._
 import sbt.internal.BuildDef
 
-import net.vonbuchholtz.sbt.dependencycheck.DependencyCheckPlugin.autoImport.dependencyCheckFailBuildOnCVSS
+import com.typesafe.sbt.SbtPgp.autoImport._
+import net.vonbuchholtz.sbt.dependencycheck.DependencyCheckPlugin.autoImport._
+import sbtrelease.ReleasePlugin.autoImport._
+import sbtrelease.ReleaseStateTransformations._
+import xerial.sbt.Sonatype.autoImport._
 
+/** Creates a dynamic project for each schema version, and configures the common tests for all projects.
+  *
+  * The main projects are cross-compiled but the tests aren't because of a test dependency that isn't available for 2.12.
+  * Releasing, publishing, and checking for insecure dependencies are disabled for test projects.
+  * The root project is just an empty shell to run aggregated commands.
+  *
+  * The release process is customised to include a step to run ''dependencyCheck''.
+  */
 object Build extends BuildDef with BuildCommon {
-  override def projects: Seq[Project] = mainProjects ++ testProjects
+  private val schemaVersions = Seq("3.3", "3.4", "3.5", "3.6")
+
   lazy val mainProjects: Seq[Project] = schemaVersions.map(nitfProject)
   lazy val testProjects: Seq[Project] = schemaVersions.map(testProject)
+  lazy val realProjects: Seq[Project] = mainProjects ++ testProjects
 
-  private val releaseVersion = 0
-  private val schemaVersions = Seq("3.3", "3.4", "3.5", "3.6")
+  override def projects: Seq[Project] = rootProject.toSeq ++ realProjects
```

Comment:
why the `Some -> Seq` kung-fu? Wouldn't `rootProject +: realProjects` just work? (assuming `rootProject: Project`)

Reply:
`rootProject` is defined on the following line as an `Option`. It's an override so I can't change its signature.

Reply:
that OO stuff ... my eyes abstract these things away 🙄